examples/AffinePlaneCurve.jl

```diff
@@ -384,14 +396,229 @@ function curve_singular_locus(C::AffinePlaneCurve)
+"""
+function ideal_point(R::MPolyRing{S}, P::Point{S}) where S <: FieldElem
+  V = gens(R)
+  return ideal(R, [V[i] - P.coord[i] for i in 1:length(V)])
+end
+
+################################################################################
+# Helping function:
+# change of variable to send a point at the origin.
+
+function curve_map_point_origin(C::AffinePlaneCurve, P::Point)
+  F = C.eq
+  R = parent(C.eq)
+  V = gens(R)
+  G = evaluate(F, [V[1] + P.coord[1], V[2] + P.coord[2]])
+  return(AffinePlaneCurve(G))
+end
+
+################################################################################
+# Helping function:
+# used to order elements to find the multiplicity.
+
+function _sort_helper_multiplicity(a::GrpAbFinGenElem)
+  return a.coeff[1, 1]
+end
+
+################################################################################
+# compute the multiplicity of the affine plane curve C at the point P
+
+@doc Markdown.doc"""
+multiplicity(C::AffinePlaneCurve{S}, P::Point{S}) where S <: FieldElem
+> Returns the multiplicity of C at P.
+"""
+function multiplicity(C::AffinePlaneCurve{S}, P::Point{S}) where S <: FieldElem
+  if P.ambient_dim != 2
+     error("The point needs to be in a two dimensional space")
+  end
+  D = curve_map_point_origin(C, P)
+  G = D.eq
+  R = parent(G)
+  A = grade(R)
+  HC = homogenous_components(A(G))
+  L = collect(keys(HC))
+  M = sort(L, by=_sort_helper_multiplicity)
+  return(M[1].coeff[1, 1])
+end
+
+################################################################################
+# compute the set of tangent lines of the affine plane curve C at the point P
+# (linear factors of the homogeneous part of lower degree of the equation).
+
+@doc Markdown.doc"""
+tangent_lines(C::AffinePlaneCurve{S}, P::Point{S}) where S <: FieldElem
+> Returns the tangent lines at P to C with their multiplicity.
+"""
+function tangent_lines(C::AffinePlaneCurve{S}, P::Point{S}) where S <: FieldElem
+  if P.ambient_dim != 2
+     error("The point needs to be in a two dimensional space")
+  end
+  D = curve_map_point_origin(C, P)
+  G = D.eq
+  R = parent(G)
+  V = gens(R)
+  A = grade(R)
+  HC = homogenous_components(A(G))
+  L = collect(keys(HC))
+  M = sort(L, by=_sort_helper_multiplicity)
+  Gm = HC[M[1]]
+  Z = factor(Gm)
+  D = Dict{AffinePlaneCurve{S}, Int64}()
+  X = V[1] - P.coord[1]
+  Y = V[2] - P.coord[2]
+  for p in keys(Z.fac)
+     if total_degree(p.f) == 1
+        push!(D, AffinePlaneCurve(evaluate(p.f, [X, Y])) => Z.fac[p])
+     end
+  end
+  return(D)
+end
+
+################################################################################
+# check for equality of curves (the equations are equal up to multiplication by
+# a non zero constant).
+
+function ==(C::AffinePlaneCurve, D::AffinePlaneCurve)
+  F = C.eq
+  G = D.eq
+  d = div(F, G)
+  if isconstant(d) == false
+     return false
+  else
+     if F - d*G == 0
+        return true
+     else
+        return false
+     end
+  end
```

Comment:
This performs a full polynomial division, doesn't it? You could instead check whether the degrees are equal, and then comput `d` just from the lead coefficients? In fact, perhaps this would do it:
```suggestion
  return degree(F) == degree(G) && F*(lc(G)/lc(F)) == G
```